src/main/java/seedu/address/model/ModelPredicate.java

```diff
@@ -22,4 +23,15 @@
         person -> person.getInfectionStatus().getStatusAsBoolean();
     public static final Predicate<Person> PREDICATE_SHOW_ALL_QUARANTINED =
         person -> person.getQuarantineStatus().getStatusAsBoolean();
+
+    /** {@code Predicate} to handle location list */
+    public static Predicate<Location> getPredicateShowLocationsByPerson(List<Integer> locationIds) {
+        return location -> {
+            boolean condition = location.getId().getZeroBased() == locationIds.get(0);
+            for (int i = 1; i < locationIds.size(); i++) {
+                condition = condition || location.getId().getZeroBased() == locationIds.get(i);
```

Comment:
Might be better to name the conditions separately with informative names as this line of code can get confusing.